mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASDeviceTest.java

```diff
@@ -82,4 +84,15 @@ public void testDeviceIdentifierAfterDeregister() throws InterruptedException, E
         Assert.assertFalse(identifier1.equals(identifier2));
     }
 
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
We need more test cases to cover all the scenarios, you have to define new mock endpoints under com.ca.mas.GatewayDefaultDispatcher to mock the response from the server.